pkg/controllers/provisioning/provisioner_test.go

```diff
@@ -1358,4 +1375,73 @@ var _ = Describe("Provisioner/Provisioning", func() {
 			})
 		})
 	})
+
+	Context("Resource Calculations", func() {
+		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers for order type1", func() {
+			always := v1.ContainerRestartPolicyAlways
+
+			pod := test.Pod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2")},
+				},
+
+				InitImage: "pause",
+
+				InitContainers: []test.InitContainersOptions{
+					{
+						RestartPolicy: &always,
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+					{
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+				},
+			})
+			podResources := resources.Ceiling(pod)
+			Expect(podResources.Limits.Cpu().CmpInt64(8)).To(Equal(0))
+			Expect(podResources.Limits.Memory().CmpInt64(4294967296)).To(Equal(0))
+
+		})
+
+		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers for order type2", func() {
```

Comment:
What is order type2? This isn't clear on the initial read.

Reply:
type1 is where we have sidecar containers at end after init sidecards and type2 is vice versa of it . its just naming the order since different order has different calculations